### PR TITLE
Versioning for ValidatorStake (with nightly version bump)

### PR DESCRIPTION
**Pull request by birchmd**
This PR re-instates the changes made in https://github.com/near/nearcore/pull/4115 with two small changes. These small changes are:
* Re-using the `Store` that is created in the nightly DB migration block
* Increasing the nightly protocol version from 108 to 109. This is necessary to trigger the hard-fork on betanet and thus avoid the issue reported in https://github.com/near/nearcore/pull/4178

For reviewers' convenience these two changes are given as separate commits, so they are isolated from the original (very large) PR and easily reviewed independently.

Testing plan: existing tests.

### CODE REVIEW

**Review by bowenwang1996 — APPROVED** [review]

**Review by frol — APPROVED** [review]

**Comment by birchmd** [comment]
@ailisp @evgenykuzyakov Please take a look, I need your approval since you are code owners.

**Review by ailisp — APPROVED** [review]

**Pull request merged**
